kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/TopicResolver.java

```diff
@@ -0,0 +1,14 @@
+package org.axonframework.extensions.kafka.eventhandling.producer;
+
+import org.axonframework.eventhandling.EventMessage;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Interface to determine if a message should be published to Kafka, and if so to which topic. If the result from the
```

Comment:
I would specify we're dealing with an `EventMesage` here instead of just "a message."